### PR TITLE
Make ForgotPassword UX slightly more user friendly

### PR DESCRIPTION
**Pull request by Palid**
Fixes https://github.com/vector-im/element-web/issues/11531

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Make ForgotPassword UX slightly more user friendly ([\#6636](https://github.com/matrix-org/matrix-react-sdk/pull/6636)). Fixes vector-im/element-web#11531. Contributed by [Palid](https://github.com/Palid).

Preview: https://611e125992fa07f14a07123e--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/structures/auth/_Login.scss [hunk 1]
```diff
@@ -96,3 +96,10 @@ div.mx_AccessibleButton_kind_link.mx_Login_forgot {
         cursor: not-allowed;
     }
 }
+.mx_Login--spinner {
```
Comment (SimonBrandner) [inline res/css/structures/auth/_Login.scss:99]:
```suggestion
.mx_Login_spinner {
```

This seems to match our naming conventions better

Comment (Palid) [reply res/css/structures/auth/_Login.scss:99]: It's all over the place, but sure. 😆

File: src/components/structures/auth/ForgotPassword.tsx [hunk 2]
```diff
@@ -357,6 +376,8 @@ export default class ForgotPassword extends React.Component<IProps, IState> {
             case Phase.Done:
                 resetPasswordJsx = this.renderDone();
                 break;
+            default:
+                resetPasswordJsx = <div className="mx_Login--spinner"><InlineSpinner w={64} h={64} /></div>;
```
Comment (SimonBrandner) [inline src/components/structures/auth/ForgotPassword.tsx:380]:
```suggestion
                resetPasswordJsx = <div className="mx_Login_spinner"><InlineSpinner w={64} h={64} /></div>;
```

File: src/components/structures/auth/ForgotPassword.tsx [hunk 3]
```diff
@@ -320,6 +336,9 @@ export default class ForgotPassword extends React.Component<IProps, IState> {
                 type="button"
                 onClick={this.onVerify}
                 value={_t('I have verified my email address')} />
+            { this.state.currentHttpRequest && (
+                <div className="mx_Login--spinner"><InlineSpinner w={64} h={64} /></div>)
```
Comment (SimonBrandner) [inline src/components/structures/auth/ForgotPassword.tsx:340]:
```suggestion
                <div className="mx_Login_spinner"><InlineSpinner w={64} h={64} /></div>)
```

File: res/css/structures/auth/_Login.scss [hunk 4]
```diff
@@ -96,3 +96,10 @@ div.mx_AccessibleButton_kind_link.mx_Login_forgot {
         cursor: not-allowed;
     }
 }
+.mx_Login_spinner {
```
Comment (gsouquet) [inline res/css/structures/auth/_Login.scss:99]: I can't find a match for that CSS class, 

https://github.com/matrix-org/matrix-react-sdk/pull/6636/files#diff-4d55f993b98a009668aad5ab3b9223bacf7c0c140cc109d508b050de07a23d94R380 and https://github.com/matrix-org/matrix-react-sdk/pull/6636/files#diff-4d55f993b98a009668aad5ab3b9223bacf7c0c140cc109d508b050de07a23d94R340 both refer to `mx_Login--spinner`

Also we could probably change the name of that CSS class, it doesn't look related to the login page nor a spinner, it's more a utility to centre items vertically and horizontally

Comment (Palid) [reply res/css/structures/auth/_Login.scss:99]: @gsouquet yeah, commited @SimonBrandner suggestion but didn't do it everywhere.

Comment (Palid) [reply res/css/structures/auth/_Login.scss:99]: @gsouquet it adds the padding though, which is very case-dependant here.

**Review by gsouquet — APPROVED** [review]

**Pull request merged**